logging/src/logging/filter.cpp

```diff
@@ -0,0 +1,507 @@
+        const auto& val = logging::find_attribute(attributes, this->attribute_name);
+        return (val && val->value.template convertible_to<T>() && val->value.template to<T>() > this->attribute_value)
+                ? fr::accept
+                : fr::reject;
+    }
+
+    virtual
+    dynamic_t
+    representation() const {
+        return dynamic_t(std::make_tuple(std::string(">"), this->attribute_name, this->attribute_value));
+    };
+};
+
+template<class T>
+struct less_filter_t: public comparision_filter_t<T> {
+    less_filter_t(std::string _attribute_name, T _attribute_value):
+        comparision_filter_t<T>(std::move(_attribute_name), std::move(_attribute_value))
+    {}
+
+    virtual
+    filter_result_t
+    apply(const std::string&, unsigned int, const logging::attributes_t& attributes) const {
+        const auto& val = logging::find_attribute(attributes, this->attribute_name);
+        return (val && val->value.template convertible_to<T>() && val->value.template to<T>() < this->attribute_value)
+                ? fr::accept
+                : fr::reject;
+    }
+
+    virtual
+    dynamic_t
+    representation() const {
+        return dynamic_t(std::make_tuple(std::string("<"), this->attribute_name, this->attribute_value));
+    };
+};
+
+template<class T>
+struct greater_or_equal_filter_t: public comparision_filter_t<T> {
+    greater_or_equal_filter_t(std::string _attribute_name, T _attribute_value):
+        comparision_filter_t<T>(std::move(_attribute_name), std::move(_attribute_value))
+    {}
+
+    virtual
+    filter_result_t
+    apply(const std::string&, unsigned int, const logging::attributes_t& attributes) const {
+        const auto& val = logging::find_attribute(attributes, this->attribute_name);
+        return (val && val->value.template convertible_to<T>() && val->value.template to<T>() >= this->attribute_value)
+                ? fr::accept
+                : fr::reject;
+    }
+
+    virtual
+    dynamic_t
+    representation() const {
+        return dynamic_t(std::make_tuple(std::string(">="), this->attribute_name, this->attribute_value));
+    };
+};
+
+template<class T>
+struct less_or_equal_filter_t: public comparision_filter_t<T> {
+    less_or_equal_filter_t(std::string _attribute_name, T _attribute_value):
+        comparision_filter_t<T>(std::move(_attribute_name), std::move(_attribute_value))
+    {}
+
+    virtual
+    filter_result_t
+    apply(const std::string&, unsigned int, const logging::attributes_t& attributes) const {
+        const auto& val = logging::find_attribute(attributes, this->attribute_name);
+        return (val && val->value.template convertible_to<T>() && val->value.template to<T>() <= this->attribute_value)
+                ? fr::accept
+                : fr::reject;
+    }
+
+    virtual
+    dynamic_t
+    representation() const {
+        return dynamic_t(std::make_tuple(std::string("<="), this->attribute_name, this->attribute_value));
```

Comment:
Did you think that all your representations are just tuple of three tokens? Looks like duplicate for me.